### ISSUE TITLE
Project Tracking using Github

### ISSUE THREAD

**Issue by okkhoy** [nus-cs2103-AY1718S2/forum#28]
# Project Tracking using Github

1. Create issues for each user story that is recorded; these issues are labeled `type.story`
     - Minimally should create issues for each user story that is marked as high priority

2. From the set of user stories, decide on the features. Create an issue for each feature you plan to implement; these issues are labeled `type.epic`. 
    - A `type.epic` issue is a big feature and can be simplified to smaller stories. 

3. Issues are created for tangible tasks. Tasks are the things that need to be done. These issues are labeled as `type.task`
    - It is encouraged to create multiple `type.task` issues for each `type.story` issue for tracking purpose.
    - You can reference other issues using the `#<issue number>` tag when you are creating a new issue, so that there is a clear trail for the feature that you plan to implement.

4. Create milestones on Github `v1.0` through `v1.5rc` and `v0.5`. One for each week.
    - Assign an end date/due date for each milestone to be their tutorial date. e.g., Wednesday teams would set `v1.2` as March 21, 2018

5. You can now assign milestones to each issue in Github.
    - When you select an issue and open it, on the right side of the page, you can set milestones, owner (or assignee) and labels. 
    - This allows the TA to know how many issues are being resolved in each milestone.

6. As and when you resolve an issue (i.e. complete a task),  mark the corresponding issue as **closed**. 
    - This updates the progress of the corresponding milestone.
    - Tips: This can be automated by using “Fixes `#<issue number>`” in the pull request you make to the master/version branch. GitHub will automatically close the issue referenced.

7. When all `type.task`s are closed, the referenced `type.story` and `type.epic` can be closed as well.
    - Remember to ensure the build is successful and the majority of the tests are passing.

8. When all issues for a milestone are closed:
    - You can **tag the version with the milestone number** (using `git tag` command) and create a release on Github.
    - You need to **explicitly close the milestone on Github**. This has to be done before the tutorial.

# What happens during tutorials?

- In the tutorial, tutor will provide feedback to you about the functionality, features or identify bugs during the demo.
- The teams need to record these feedback items as issues in the issue tracker and assign appropriate labels like `type.bug`, `type.enhancement` or `type.task`.
- Such issues should be assigned to some subsequent milestones, and they get tracked like other issues in the same way as mentioned above.
